### ISSUE TITLE
Breaking change in Formik validate v2

### ISSUE THREAD

**Issue by gwn** [jaredpalmer/formik#2167]
## 📖 Documentation
Validate function is not accepting rejected promises of errors, as it did in v1. I was going to report this as a bug, but this looked intentional to me after checking out the code:

https://github.com/jaredpalmer/formik/blob/9316a864478f8fcd4fa99a0735b1d37afdf507dc/packages/formik/src/Formik.tsx#L190-L206

So if this is intentional, it needs to be documented among other breaking changes (which atm is not):
https://github.com/jaredpalmer/formik/blob/master/docs/migrating-v2.md#breaking-changes

If not, needs to be fixed.

Related issue from v1: https://github.com/jaredpalmer/formik/issues/966

**Comment by jaredpalmer** [comment]
Yeah. Shit, I guess this is breaking after we stopped swallowing the errors in recent releases. can you add to the migration docs?

**Comment by gwn** [comment]
Sure, I will.

**Comment by jaredpalmer** [comment]
Thanks!
